jinahub/segmenters/VADSpeechSegmenter/vad_speech_segmenter.py

```diff
@@ -0,0 +1,135 @@
+__copyright__ = 'Copyright (c) 2020-2021 Jina AI Limited. All rights reserved.'
+__license__ = 'Apache-2.0'
+
+from pathlib import Path
+from typing import Optional, Dict
+
+import torch
+import torchaudio
+import numpy as np
+import librosa as lr
+from jina import Executor, Document, DocumentArray, requests
+from jina.excepts import BadDocType
+
+
+class VADSpeechSegmenter(Executor):
+    """
+     Segment the speech audio using Silero's Voice Activity Detector (VAD)
+
+    :param normalize: a bool to specify whether to normalize the audio
+     by the sample rate
+    :param dump: a bool to specify whether to dump the segmented audio
+    :param repo: default repo name of silero-vad
+    :param model: default model name of silero-vad
+
+    """
+
+    TARGET_SAMPLE_RATE = 16000
+
+    def __init__(
+        self,
+        normalize: bool = False,
+        dump: bool = False,
+        repo: str = 'snakers4/silero-vad',
+        model: str = 'silero_vad',
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        if model is None or repo is None:
+            raise ValueError('model and repo cannot be None')
+
+        # TODO: remove the following temporary fix for torch hub
+        torch.hub._validate_not_a_forked_repo = lambda *args: True
+        self.model, utils = torch.hub.load(
+            repo_or_dir=repo, model=model, force_reload=True
+        )
+
+        (_, self.get_speech_ts_adaptive, self.save_audio, *_) = utils
+        self.normalize = self._normalize if normalize else lambda audio, _: audio
+        self.dump = dump
+
+    @requests
+    def segment(
+        self,
+        docs: Optional['DocumentArray'] = None,
+        parameters: Optional[Dict] = {},
```

Comment:
```suggestion
        parameters: dict = {},
```
please check the python doc to identify how to use `Optional`

Reply:
Yes, `Optional` should not be used in this case because the default is `{}` not `None`. `Optional` is equivalent to `Union[...,None]`. I have addressed this [here](https://github.com/jina-ai/executors/pull/97/commits/752f958d7e44f07d635699131e1a7a43febce5ab), thank you.